tests/utilities/test_apply_func_torchtext.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import torch
+import torchtext
+from torchtext.data.example import Example
+
+from pytorch_lightning.utilities.apply_func import move_data_to_device
+
+
+def _get_torchtext_data_iterator(include_lengths=False):
+    text_field = torchtext.data.Field(sequential=True, pad_first=False,  # nosec
+                                      init_token="<s>", eos_token="</s>",  # nosec
+                                      include_lengths=include_lengths)  # nosec
+
+    example1 = Example.fromdict({"text": "a b c a c"}, {"text": ("text", text_field)})
+    example2 = Example.fromdict({"text": "b c a a"}, {"text": ("text", text_field)})
+    example3 = Example.fromdict({"text": "c b a"}, {"text": ("text", text_field)})
+
+    dataset = torchtext.data.Dataset([example1, example2, example3],
+                                     {"text": text_field}
+                                     )
+    text_field.build_vocab(dataset)
+
+    iterator = torchtext.data.Iterator(dataset, batch_size=3,
+                                       sort_key=None, device=None,
+                                       batch_size_fn=None,
+                                       train=True, repeat=False, shuffle=None,
+                                       sort=None, sort_within_batch=None)
+    return iterator, text_field
+
+
+@pytest.mark.parametrize('include_lengths', [False, True])
+@pytest.mark.parametrize(['device'], [pytest.param(torch.device('cuda', 0))])
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="test assumes GPU machine")
+def test_batch_move_data_to_device_torchtext_include_lengths(include_lengths, device):
+    data_iterator, _ = _get_torchtext_data_iterator(include_lengths=include_lengths)
+    data_iter = iter(data_iterator)
+    batch = next(data_iter)
+    batch_on_device = move_data_to_device(batch, device)
+
+    if include_lengths:
+        # tensor with data
+        assert (batch_on_device.text[0].device == device)
+        # tensor with length of data
+        assert (batch_on_device.text[1].device == device)
+    else:
+        assert (batch_on_device.text.device == device)
+
+
+@pytest.mark.parametrize('include_lengths', [False, True])
+def test_batch_move_data_to_device_torchtext_include_lengths_cpu(include_lengths):
+    test_batch_move_data_to_device_torchtext_include_lengths(include_lengths, torch.device('cpu'))
```

Comment:
this can also be included in the above test parameter, right?

Reply:
Thanks. I tried that first, but if the machine has no GPU it failed the test, and if I skip the above test if no GPU is available it does not run on a CPU either. That is why I split it into two. I had a "smart" way to create a list of devices conditioned on on if a GPU is available, but found that I got confused when the numbers of test run varied from machine to machine. So I opted for this simple and easy to understand approach.

Reply:
> I tried that first, but if the machine has no GPU it failed the test, and if I skip the above test if no GPU is available it does not run on a CPU either.

That is bad, it means there is something wrong with this fix or the test. Which commit was that, can I see the error message? setting the device=cpu on a machine that has gpu should definitely work

Reply:
I added a commit, I am convinced it has to work. Anyway, it's just a cosmetic change to get rid of the last test method. But feel free to undo if it does not work.

Reply:
I used the following construct which only runs the test on a CPU and GPU if a GPU is available otherwise only uses CPU:
`
@pytest.mark.parametrize(['device'],
                         [pytest.param(torch.device('cuda', 0)),
                          pytest.param(torch.device('cpu'))] if torch.cuda.is_available() else [
                             pytest.param(torch.device('cpu'))])
@pytest.mark.parametrize('include_lengths', [False, True])

`